src/storage/btree.h

```diff
@@ -192,6 +210,19 @@ struct btree_scan
   DB_VALUE cur_key;		/* current key value */
   bool clear_cur_key;		/* clear flag for current key value */
 
+  //---------------------------------------------------------------------------------------------   
+  /* IMPROVE_RANGE_SCAN_IN_BTREE */
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+  VPID cur_common_prefix_page_vpid;
+  LOG_LSA cur_common_prefix_page_lsa;
+#endif
+  int common_prefix_size;	// size of common prefix key parts.
+  DB_VALUE common_prefix_key;	// common prefix key part 
+  bool clear_common_prefix_key;	// 
+  bool is_cur_key_compressed;	/* If false, cur_key is a complete key.
```

Comment:
I think it can be differentiated by common_prefix_size, but is is_cur_key_compressed needed?
  - If -1, unknown
  - If 0, complete key
  - If it is greater than 0, there is a prefix key

Reply:
is_cur_key_compressed indicates the state of cur_key, and common_prefix_size is a value for the state of the page.
That is, regardless of common_prefix_size, cur_key may be compressed or uncompressed.

Reply:
When cur_key changes to the uncompressed state, if common_prefix_size is also changed to 0, isn't is_cur_key_compressed unnecessary?

Reply:
I understood that the life cycle of common_prefix_size and is_cur_key_compressed changing is different.
If common_prefix_size must remain the same within the same page even if cur_key changes, is_cur_key_compressed is needed.